src/dotnet-install.sh

```diff
@@ -290,18 +290,42 @@ combine_paths() {
     return 0
 }
 
+get_shell_bitness() {
```

Comment:
As I said previously I'm not in favor of this workaround. 
There might be other things in dotnet SDK or runtime which are relying on arch, and if OS reports it wrongly this may lead to unexpected results. However, I'm not an expert in this area, so cannot really advise and assess the consequences.